gen3users/validation.py

```diff
@@ -565,3 +566,74 @@ def validate_users(user_yaml_dict, existing_resources):
             #     )
 
     return ok
+
+
+def validate_roles(user_yaml_dict):
+    """
+    Validates the "roles" section of the user.yaml by checking that all
+    required fields are present.
+
+    Args:
+        user_yaml_dict (dict): Contents of a user.yaml file.
+
+    Return:
+        ok(bool): whether the validation succeeded.
+    """
+    logger.info("- Validating roles")
+    ok = True
+
+    role_ids = get_field_from_list(user_yaml_dict["authz"].get("roles", []), "id")
```

Comment:
`get_field_from_list` iterates through the whole list so i would rather avoid using it multiple times like this. In other tests, `get_field_from_list` is used when we want the list of values but we won't iterate over it.
Here since we need to manually iterate through the whole list anyway (line 605), i would do simplify this to something like:
```
for role in user_yaml_dict["authz"].get("roles", []):
    check there's an "id"
    check there are "permissions"
    for each permission:
        ...
        for each action:
            ...
```
this will also let you output a more useful message, like "role X is missing Y" instead of "a role is missing Y"